midterm/script.js

```diff
@@ -0,0 +1,83 @@
+(async() => {
+	'use strict';
+
+  const URL = 'https://54o76ppvn8.execute-api.ap-northeast-1.amazonaws.com/prod/bb-dojo';
+  const Limit = 'limit 5';
+  const getNewsInfo = await axios.get(URL, {
+    params: {
+      id: 'dojo',
+      query: Limit
+    }
+  }).catch((err) => {
+    console.log('err', err);
+  });
+  
+  //取得したレスポンスのdataのみを抽出
+  const newsData = getNewsInfo.data;
+
+  //main > tbl > tblBody の構造を持つように要素を追加
+  const main = document.getElementsByTagName('main')[0];
+  const tbl = document.createElement("table");
+  const tblBody = document.createElement("tbody");
+  
+  //５件取得してきたnewsDataを一件ずつの処理に置き換えて、表を作成する
+  const row = Object.keys(newsData).forEach((key) => {
+    const tblRow = document.createElement('tr');
+    
+    //td要素の作成と日付の書き込み
+    const tblDayData = document.createElement('td');
+    tblDayData.textContent = newsData[key].day.value;
+    
+    //td要素の作成とカテゴリの書き込み
+    const tblCategoryData = document.createElement('td');
+    tblCategoryData.textContent = newsData[key].category.value;
+    
+    //categoryによる背景色の条件分岐
+    const label = newsData[key].label.value;
+    switch (label){
+      case 'company':
+        tblCategoryData.classList.add('backcolor_company');
+        break;
+        case 'product':
+        tblCategoryData.classList.add('backcolor_product');
+        break;
+        case 'ir':
+        tblCategoryData.classList.add('backcolor_ir');
+        break;
+    };
+
+    //td要素の作成
+    const tblContentData = document.createElement('td');
+
+    //a要素の作成(リンクのため)
+    const hrefInfo = document.createElement('a');
+    hrefInfo.href = newsData[key].url.value; //url.valueに格納されたURLを参照
+    hrefInfo.target = '_blank'; //別ウィンドウで開く
```

Comment:
`newsData` オブジェクトの中身を見ていくと `target` というキーがあるので、それの値を利用してください！
```suggestion
    hrefInfo.target = hrefInfo.target = key.target.value;
```

（本来はドメインが異なるサイトに移動する場合は別タブで開いたほうがよいのですが、今回は課題資料4ページ目に「targetが_selfなら同じウィンドウ、_blankなら別ウィンドウで開かせること」とあるので）